<a href="https://colab.research.google.com/github/HalimahOye/food_balance_sheet/blob/main/food_balance_sheet.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd

In [ ]:
df = pd.read_csv("FoodBalanceSheets_E_Africa_NOFLAG.csv", encoding ="latin-1")

In [ ]:
df.head()

Area Code     Area  Item Code         Item  Element Code  \
0          4  Algeria       2501   Population           511   
1          4  Algeria       2501   Population          5301   
2          4  Algeria       2901  Grand Total           664   
3          4  Algeria       2901  Grand Total           674   
4          4  Algeria       2901  Grand Total           684   

                                  Element             Unit     Y2014  \
0           Total Population - Both sexes     1000 persons  38924.00   
1                Domestic supply quantity      1000 tonnes      0.00   
2           Food supply (kcal/capita/day)  kcal/capita/day   3377.00   
3  Protein supply quantity (g/capita/day)     g/capita/day     94.90   
4      Fat supply quantity (g/capita/day)     g/capita/day     80.06   

      Y2015     Y2016     Y2017     Y2018  
0  39728.00  40551.00  41389.00  42228.00  
1      0.00      0.00      0.00      0.00  
2   3379.00   3372.00   3341.00   3322.00  
3     94.35     94.72     92.82     91.83  
4     79.36     77.40     80.19     77.28

In [ ]:
df.shape

(60943, 12)

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 60943 entries, 0 to 60942
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Area Code     60943 non-null  int64  
 1   Area          60943 non-null  object 
 2   Item Code     60943 non-null  int64  
 3   Item          60943 non-null  object 
 4   Element Code  60943 non-null  int64  
 5   Element       60943 non-null  object 
 6   Unit          60943 non-null  object 
 7   Y2014         59354 non-null  float64
 8   Y2015         59395 non-null  float64
 9   Y2016         59408 non-null  float64
 10  Y2017         59437 non-null  float64
 11  Y2018         59507 non-null  float64
dtypes: float64(5), int64(3), object(4)
memory usage: 5.6+ MB


In [ ]:
#check for missing values
df.isnull().sum()

Area Code          0
Area               0
Item Code          0
Item               0
Element Code       0
Element            0
Unit               0
Y2014           1589
Y2015           1548
Y2016           1535
Y2017           1506
Y2018           1436
dtype: int64

In [ ]:
df.describe()



Area Code     Item Code  Element Code          Y2014          Y2015  \
count  60943.000000  60943.000000  60943.000000   59354.000000   59395.000000   
mean     134.265576   2687.176706   3814.856456     134.196282     135.235966   
std       72.605709    146.055739   2212.007033    1567.663696    1603.403984   
min        4.000000   2501.000000    511.000000   -1796.000000   -3161.000000   
25%       74.000000   2562.000000    684.000000       0.000000       0.000000   
50%      136.000000   2630.000000   5142.000000       0.090000       0.080000   
75%      195.000000   2775.000000   5511.000000       8.340000       8.460000   
max      276.000000   2961.000000   5911.000000  176405.000000  181137.000000   

               Y2016          Y2017          Y2018  
count   59408.000000   59437.000000   59507.000000  
mean      136.555222     140.917765     143.758381  
std      1640.007194    1671.862359    1710.782658  
min     -3225.000000   -1582.000000   -3396.000000  
25%         0.000000       0.000000       0.000000  
50%         0.080000       0.100000       0.070000  
75%         8.430000       9.000000       9.000000  
max    185960.000000  190873.000000  195875.000000

In [ ]:
df.groupby("Y2017")["Area"].sum()

Y2017
-1582.0                       EgyptEgypt
-1200.0                       Mozambique
-1159.0                       Mozambique
-865.0       United Republic of Tanzania
-860.0       United Republic of Tanzania
                        ...             
 96443.0                           Egypt
 106400.0                       Ethiopia
 112625.0                        Nigeria
 117702.0                        Nigeria
 190873.0                        Nigeria
Name: Area, Length: 4537, dtype: object

In [ ]:
df.corr()["Element Code"]

<ipython-input-9-4e3252ccc62f>:1: FutureWarning: The default value of numeric_only in DataFrame.corr is deprecated. In a future version, it will default to False. Select only valid columns or specify the value of numeric_only to silence this warning.
  df.corr()["Element Code"]


Area Code      -0.000209
Item Code      -0.024683
Element Code    1.000000
Y2014           0.024457
Y2015           0.023889
Y2016           0.023444
Y2017           0.024254
Y2018           0.024279
Name: Element Code, dtype: float64

2016 has the least correlation with element code


In [ ]:
df.groupby("Item")["Y2015","Y2018"].sum()

<ipython-input-10-0498afde6108>:1: FutureWarning: Indexing with multiple keys (implicitly converted to a tuple of keys) will be deprecated, use a list instead.
  df.groupby("Item")["Y2015","Y2018"].sum()


Y2015      Y2018
Item                                     
Alcohol, Non-Food      2180.00    2293.00
Alcoholic Beverages   98783.72   97847.27
Animal Products       11811.73   11578.61
Animal fats          200675.72  269648.27
Apples and products   10559.15    9640.51
...                        ...        ...
Vegetables, Other    158104.08  163987.21
Vegetal Products     107064.17  107775.39
Wheat and products   234710.51  242645.19
Wine                   4251.81    4039.32
Yams                 203151.78  221272.09

[119 rows x 2 columns]

Total Number of Wine for "Y2015" and "Y2018" respectively are
4251.81 and 4039.32


In [ ]:
df["Area"].nunique()

49

the total unique country is 49

In [23]:
df.groupby(["Area","Element"])["Y2015"].sum() # groupby multiple columns

Area      Element                               
Algeria   Domestic supply quantity                  85972.86
          Export Quantity                            1104.62
          Fat supply quantity (g/capita/day)          317.40
          Feed                                      10308.00
          Food                                      61693.49
                                                      ...   
Zimbabwe  Protein supply quantity (g/capita/day)      175.31
          Residuals                                  -780.00
          Seed                                        128.00
          Stock Variation                            -343.00
          Total Population - Both sexes             13815.00
Name: Y2015, Length: 803, dtype: float64

In [24]:
multiple_aggregations = df.groupby(['Area', 'Element'])['Y2015'].agg(['count', 'sum', 'mean'])

In [26]:
multiple_aggregations

count       sum          mean
Area     Element                                                              
Algeria  Domestic supply quantity                  113  85972.86    760.821770
         Export Quantity                            76   1104.62     14.534474
         Fat supply quantity (g/capita/day)        107    317.40      2.966355
         Feed                                       20  10308.00    515.400000
         Food                                      104  61693.49    593.206635
...                                                ...       ...           ...
Zimbabwe Protein supply quantity (g/capita/day)    112    175.31      1.565268
         Residuals                                 104   -780.00     -7.500000
         Seed                                       19    128.00      6.736842
         Stock Variation                            95   -343.00     -3.610526
         Total Population - Both sexes               1  13815.00  13815.000000

[803 rows x 3 columns]

The protein supply quantity for Madagascar is 173.05

In [ ]:
df.groupby("Element")["Y2014","Y2015","Y2016","Y2017","Y2018"].sum()


<ipython-input-12-0df15587a8ba>:1: FutureWarning: Indexing with multiple keys (implicitly converted to a tuple of keys) will be deprecated, use a list instead.
  df.groupby("Element")["Y2014","Y2015","Y2016","Y2017","Y2018"].sum()


Y2014       Y2015       Y2016  \
Element                                                                      
Domestic supply quantity                1996716.35  2021493.55  2044842.70   
Export Quantity                          150020.64   157614.47   151920.46   
Fat supply quantity (g/capita/day)        10225.56    10235.74    10102.77   
Feed                                     216927.89   225050.22   228958.65   
Food                                    1212332.49  1232361.10  1247022.17   
Food supply (kcal/capita/day)            454257.00   453383.00   451810.00   
Food supply quantity (kg/capita/yr)       49650.63    49345.13    48985.28   
Import Quantity                          274144.48   267018.46   286582.78   
Losses                                   153223.00   155439.00   157787.00   
Other uses (non-food)                     78718.13    66254.41    69563.68   
Processing                               282923.00   287929.00   280631.00   
Production                              1931287.75  1947019.39  1943537.15   
Protein supply quantity (g/capita/day)    11836.46    11833.95    11779.69   
Residuals                                 30149.00    30045.00    37224.00   
Seed                                      21922.92    23976.82    23389.20   
Stock Variation                           58749.83    34910.99    33140.12   
Total Population - Both sexes           1031585.00  1058081.00  1085107.00   
Tourist consumption                         416.00      349.00       89.00   

                                             Y2017       Y2018  
Element                                                         
Domestic supply quantity                2088198.10  2161192.10  
Export Quantity                          182338.80   181594.80  
Fat supply quantity (g/capita/day)        10253.84    10258.69  
Feed                                     223705.68   233489.68  
Food                                    1258888.28  1303841.28  
Food supply (kcal/capita/day)            454681.00   455261.00  
Food supply quantity (kg/capita/yr)       48690.04    49056.85  
Import Quantity                          294559.09   287997.09  
Losses                                   160614.00   163902.00  
Other uses (non-food)                     91645.97    91300.97  
Processing                               292836.00   308429.00  
Production                              2030056.89  2075072.89  
Protein supply quantity (g/capita/day)    11842.45    11833.56  
Residuals                                 35500.00    34864.00  
Seed                                      24870.14    25263.14  
Stock Variation                           54316.91    20577.91  
Total Population - Both sexes           1112641.00  1140605.00  
Tourist consumption                          91.00       90.00

The year with the highest stock variation is 2014

In [ ]:
df.groupby("Element")["Y2017"].sum()

Element
Domestic supply quantity                  2088198.10
Export Quantity                            182338.80
Fat supply quantity (g/capita/day)          10253.84
Feed                                       223705.68
Food                                      1258888.28
Food supply (kcal/capita/day)              454681.00
Food supply quantity (kg/capita/yr)         48690.04
Import Quantity                            294559.09
Losses                                     160614.00
Other uses (non-food)                       91645.97
Processing                                 292836.00
Production                                2030056.89
Protein supply quantity (g/capita/day)      11842.45
Residuals                                   35500.00
Seed                                        24870.14
Stock Variation                             54316.91
Total Population - Both sexes             1112641.00
Tourist consumption                            91.00
Name: Y2017, dtype: float64

Total sum of processing in "Y2017" was 292836.00

In [ ]:
df.groupby("Area")["Y2017"].sum().sort_values()

Area
Sudan (former)                       0.00
Ethiopia PDR                         0.00
Comoros                             59.84
Seychelles                         442.34
Sao Tome and Principe            12662.63
Cabo Verde                       14650.74
Guinea-Bissau                    19102.77
Lesotho                          21267.96
Botswana                         22101.30
Djibouti                         22729.91
Gambia                           23154.18
Gabon                            27979.64
Liberia                          29342.20
Namibia                          29874.89
Central African Republic         29937.00
Congo                            41181.68
Togo                             49841.88
Mauritius                        51114.83
Eswatini                         54343.33
Sierra Leone                     55311.33
Chad                             71594.68
Rwanda                           73663.69
Zimbabwe                         75919.34
Senegal                      

"Guinea-Bissau" is the 7th least Area in "Y2017"